File: CompanyBranchAPI/CompanyBranchCore/Entities/Branch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CompanyBranchCore.Entities
{
    public class Branch
    {
        public int Id { get; set; }
        public string Name { get; set; } 
        public string Location { get; set; }
        public bool IsDeleted { get; set; }
        public int CompanyId { get; set; }
        public Company Company { get; set; } 
    }
}

File: CompanyBranchUi/company-branch-app/src/app/models/branch.ts
export class  Branch {
  id: number | undefined;
  name: string | undefined;
  location: string | undefined;
  isDeleted: boolean | undefined;
  companyId: number | undefined;
  company: string | undefined;
  } 
  export class CreateBranch {
    id: number | undefined;
    name: string | undefined;
    location: string | undefined;
    isDeleted: boolean | undefined;
    companyId: number | undefined;
    } 
 
 
 export class  Company {
    id: number | undefined;
  name: string | undefined;
  address: string | undefined;
  isDeleted: boolean | undefined;
  branch?: Branch | null | undefined;
  }